consumers/src/unit-test/java/uk/gov/moj/sdt/consumers/ConsumerGatewayTest.java

```diff
@@ -0,0 +1,173 @@
+package uk.gov.moj.sdt.consumers;
+
+import java.math.BigInteger;
+import java.net.ConnectException;
+import java.net.SocketTimeoutException;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.mockito.stubbing.Answer;
+import uk.gov.moj.sdt.consumers.api.IIndividualRequestConsumer;
+import uk.gov.moj.sdt.consumers.exception.OutageException;
+import uk.gov.moj.sdt.consumers.exception.SoapFaultException;
+import uk.gov.moj.sdt.consumers.exception.TimeoutException;
+import uk.gov.moj.sdt.ws._2013.sdt.baseschema.StatusCodeType;
+import uk.gov.moj.sdt.ws._2013.sdt.baseschema.StatusType;
+import uk.gov.moj.sdt.ws._2013.sdt.targetapp.submitqueryresponseschema.SubmitQueryResponseType;
+
+import javax.xml.soap.SOAPException;
+import javax.xml.ws.WebServiceException;
+import javax.xml.ws.soap.SOAPFaultException;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.*;
+
+/**
+ * Test class for the consumer gateway.
+ *
+ * @author Mark Dathorne
+ */
+@ExtendWith(MockitoExtension.class)
+class ConsumerGatewayTest extends SubmitQueryConsumerTest {
+
+    @Mock
+    IIndividualRequestConsumer individualRequestConsumer;
+
+    ConsumerGateway consumerGateway;
+
+    /**
+     * Method to do any pre-test set-up.
+     */
+    @BeforeEach
+    @Override
+    public void setUp() {
+        MockitoAnnotations.openMocks(this);
+
+        submitQueryConsumer = new SubQueryConsumer();
+        submitQueryConsumer.setTransformer(mockTransformer);
+        // submitQueryConsumer.setRethrowOnFailureToConnect (true);
```

Comment:
Remove this line if it isn't needed.

Reply:
It's from the original codebase. But ok.